src/client.ts

```diff
@@ -500,12 +500,12 @@ export class SubscriptionClient {
       this.clearMaxConnectTimeout();
       this.closedByUser = false;
       this.eventEmitter.emit(this.reconnecting ? 'reconnecting' : 'connecting');
-
       const payload: ConnectionParams = typeof this.connectionParams === 'function' ? this.connectionParams() : this.connectionParams;
 
-      // Send CONNECTION_INIT message, no need to wait for connection to success (reduce roundtrips)
-      this.sendMessage(undefined, MessageTypes.GQL_CONNECTION_INIT, payload);
-      this.flushUnsentMessagesQueue();
+      Promise.resolve(payload).then((res) => {
+        this.sendMessage(undefined, 'connection_init', res);
+        this.flushUnsentMessagesQueue();
+      }).catch(console.error);
```

Comment:
How best to handle this error? I know there have been other requests to externalize error handling? Should we close and reconnect?

Reply:
Reconnecting is an option. Why not just throw the error upstream?

Reply:
Before it was just thrown...

Reply:
The behavior changes from a thrown exception to an uncaught rejected promise. Looking at the code though we don't handle exceptions in other handlers right now. I would remove the console.error though and leave it as an uncaught to be addressed in another MR.

Reply:
Wrong, you wont get unhandled promise rejection because of the catch. Removing the console the user might not know what the problem was...